drink/test-macro/src/bundle_providing.rs

```diff
@@ -0,0 +1,85 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use convert_case::{Case, Casing};
+use proc_macro2::{Ident, Span, TokenStream as TokenStream2};
+use quote::quote;
+use syn::ItemEnum;
+
+pub struct BundleProviderGenerator {
+    root_contract_name: Option<String>,
+    bundles: HashMap<String, PathBuf>,
+}
+
+impl BundleProviderGenerator {
+    pub fn new<I: Iterator<Item = (String, PathBuf)>>(
+        bundles: I,
+        root_contract_name: Option<String>,
+    ) -> Self {
+        let root_contract_name = root_contract_name.map(|name| name.to_case(Case::Pascal));
+        let bundles = HashMap::from_iter(bundles.map(|(name, path)| {
+            let name = name.to_case(Case::Pascal);
+            (name, path)
+        }));
+
+        if let Some(root_contract_name) = &root_contract_name {
+            assert!(
+                bundles.contains_key(root_contract_name),
+                "Root contract must be part of the bundles"
+            );
+        }
+
+        Self {
+            root_contract_name,
+            bundles,
+        }
+    }
+
+    pub fn generate_bundle_providing(&self, enum_item: ItemEnum) -> TokenStream2 {
```

Comment:
This is extremely nitpicky but I'd prefer `s/providing/provision/g` :smile:

Reply:
sure, done!